app/next-client-app/lib/api/request.ts

```diff
@@ -40,12 +40,11 @@ const request = async <T>(url: string, options: RequestOptions = {}) => {
       try {
         const errorResponse = await response.json();
         errorMessage = errorResponse.detail || errorMessage;
+        return `Error: ${errorResponse.detail}`;
```

Comment:
For this, it being a string makes the code that calls the request a little unclean. 
Instead - how about we return an object?

```
return { success: false, error: errorMessage };
```

And then the calling code is:

```
resp = await updateScanReport(id, "hidden", !hidden);
if (!resp.success) {
    toast.error()
}
```